pkg/ssh/ssh_keys.go

```diff
@@ -0,0 +1,112 @@
+package ssh
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/cli/cli/v2/internal/config"
+	"github.com/cli/cli/v2/internal/run"
+	"github.com/cli/safeexec"
+)
+
+type SshContext struct {
+	ConfigDir string
+	KeygenExe string
+}
+
+type SshKeyPair struct {
+	PublicKeyPath  string
+	PrivateKeyPath string
+}
+
+func (c *SshContext) LocalPublicKeys() ([]string, error) {
+	sshDir, err := c.sshDir()
+	if err != nil {
+		return nil, err
+	}
+
+	return filepath.Glob(filepath.Join(sshDir, "*.pub"))
+}
+
+func (c *SshContext) HasKeygen() bool {
+	_, err := c.findKeygen()
+	return err == nil
+}
+
+func (c *SshContext) GenerateSSHKey(keyName string, errorOnExists bool, promptPassphrase func() (string, error)) (SshKeyPair, error) {
```

Comment:
These params feel a little bloated, would an `options` type struct be preferable? The reason I didn't do it is that it's hard to make the zero-value for the options be useful, because the `keyName` is required. I guess `keyName` could still be a parameter on its own and the other options could be wrapped

Reply:
yeah i would vote to keep `keyName` as an arg and the other values as [functional options](https://golang.cafe/blog/golang-functional-options-pattern.html)

Reply:
Cool suggestion! Updated